sim/midas/src/main/scala/midas/platform/F1Shim.scala

```diff
@@ -141,99 +133,85 @@ class F1Shim(implicit p: Parameters) extends PlatformShim {
         io.dma.ar.bits.cache,
         io.dma.ar.bits.prot,
         io.dma.ar.bits.qos,
-        io.dma.ar.bits.region,
-        io.dma.ar.bits.id,
-        io.dma.ar.bits.user)
+        io.dma.ar.bits.id)
     }
 
     when (io.dma.r.fire()) {
-      printf("[dma,rfire,%x] resp %x, data %x, last %x, id %x, user %x\n",
+      printf("[dma,rfire,%x] resp %x, data %x, last %x, id %x\n",
         cyclecount,
         io.dma.r.bits.resp,
         io.dma.r.bits.data,
         io.dma.r.bits.last,
-        io.dma.r.bits.id,
-        io.dma.r.bits.user)
+        io.dma.r.bits.id)
     }
 
-  when (io.slave(0).aw.fire()) {
-    printf("[slave,awfire,%x] addr %x, len %x, size %x, burst %x, lock %x, cache %x, prot %x, qos %x, region %x, id %x, user %x\n",
+  when (io_slave(0).aw.fire()) {
```

Comment:
Wait, does it actually work to reference io_slave before it is created? I usually get a null pointer exception if I do this. Check what happens if AXIDebugPrint is set to true.

Reply:
Ugh, probably not. i was just blindly porting this.